report/2.Analysis/EmbeddedPlatforms.tex

```diff
@@ -0,0 +1,78 @@
+	\item Be able to interface with motors, most likely Lego.
+	\item Potentially be able to train a machine learning model, which is used for the prediction on which the targeting is based.
+\end{enumerate}
+
+As image recognition is an expensive computation, the current idea is to offload this to an external server or computer, which will compute the current object position and send that to the embedded device.
+
+The current idea is to use the Lego Mindstorms platform for the robotics part of the solution, and use a single-board computer for image object tracking and the machine learning model training.
+
+\subsection{Lego Mindstorms}
+Lego Mindstorms is a programmable computer kit, made by Lego.
+The platform is based around a central control computer, called the Intelligent Brick.
+It also includes a variety of sensors, motors and connection cables.
+Each generation further improves on the concept, adding more computational power and better sensors.
+
+The Lego Mindstorm platform, comes with a graphical drag and drop interface.
+However, various different compilers for many different languages exist.
+As such the choice between different generations, does not come down to language preferences.
+
+The choices for the 5th semester are the second generation, the Lego Mindstorms NXT 2.0 kit and the third generation, the Lego Mindstorms EV3 kit.
+
+\subsubsection{Lego Mindstorms NXT 2.0}
+The Lego Mindstorms NXT 2.0 (NXT) is the second generation Lego Mindstorms kit.
+The NXT Intelligent Brick is a micro computer based on the 32-bit ARM7 microprocessor, with 256 kbytes of flash memory and 64 kbytes of RAM\cite{nxt2userguide}.
+It includes a variety of sensors, such as touch, sound, light and ultrasonic (vision) sensors.
+On top of this, the consumer kit also includes servo motors and lamps.
+
+A potential major disadvantage of the NXT 2.0, is that the platform is rather old.
+This means that problems could arise, especially in regards to compatability with newer platforms.
+This will be further elaborated on in a later section
+
+\subsubsection{Lego Mindstorms EV3}
+The  Lego Mindstorms EV3 (EV3) is the third generation of the Lego Mindstorms line.
+The EV3 mainly improves on the specifications of the second generation, adding a significantly more powerful 300MHz ARM9 processor, running Linux, 16 MB of flash memory, which is extendable with a microSDHC card and 64MB of RAM\cite{ev3userguide}.
+On top of this, it also adds remote control and WiFi capabilities, as well as more recently updated compilers.
+
+The primary advantage of the EV3 is that its software is up to date and the power it brings to the table.
+The power is however also its primary disadvantage.
+As the goal of this project is to design an embedded real time system, a too powerful computer, could potentially mask some of the challenges that normally come with the development of an embedded system.
+
+%Don't know how relevant this will actually be.
```

Comment:
Grunden til jeg ville have det var at det giver mening at sige "her er de 3 største platforme. Mindstorm. Raspberry. Arduino. Det er 3 rimelig forskellige indgange til samme problemstilling. Mindstorm findes i 2 relevante varianter: NXT og EV3. De kan X. De bliver supplied af uni og har nogle lette hardware integrationer i form af stik. raspberry er næsten for stærkt til et godt RTS projekt, og har GPIO porte som let kræver en del hardware arbejde. Arduino er ret svag så god til RTS, men supporter ikke særlig mange sprog, og har irriterende GPIO porte. Derfor giver det mening at gå med LEGO, da de er supplied og er lette at arbejde med i en software kontekst uden at spilde energi på hardware arbejde. Derudover giver det mest mening at bruge NXT da den er svagere, og det dermed bliver et mere interessant RTS projekt." i et senere kapitel snakker vi så om "tracking of a target" hvor vi siger "video feed giver mening" men nævner "da vi bruger en NXT er det fanme svært at lave de udregner derpå. derfor bruger vi en rasp men behandler den, i NXT kontekst, som en sensor så den blot sender lokations data råt dertil".